lib/iris/analysis/__init__.py

```diff
@@ -2012,18 +2012,39 @@ def _compute_shared_coords(self):
                 groupby_bounds.append((key_slice.start, key_slice.stop-1))
 
         # Create new shared bounded coordinates.
-        for coord in self._shared_coords:
+        for coord, dim in self._shared_coords:
             if coord.points.dtype.kind in 'SU':
                 if coord.bounds is None:
                     new_points = []
                     new_bounds = None
+                    # np.apply_along_axis does not work with str.join, so we
+                    # need to loop through the array directly. First move axis
+                    # of interest to trailing dim and flatten the others.
+                    work_arr = np.moveaxis(coord.points, dim, -1)
+                    shape = work_arr.shape
+                    if coord.ndim == 1:
+                        work_shape = (1, ) + work_arr.shape
+                        new_shape = (len(self),)
+                    else:
+                        work_shape = (np.product(work_arr.shape[:-1]),
+                                      work_arr.shape[-1])
```

Comment:
If I've understood this correctly, I think `work_shape = (-1, work_arr.shape[-1])` will work for both branches.

Reply:
Nice!